src/Integration.Vsix/CFamily/CMake/CMakeRequestFactory.cs

```diff
@@ -58,7 +59,7 @@ public async Task<IRequest> TryCreateAsync(string analyzedFilePath, CFamilyAnaly
             }
 
             var rulesConfig = rulesConfigProvider.GetRulesConfiguration(languageKey);
-            var context = new RequestContext(languageKey, rulesConfig, analyzedFilePath, SubProcessFilePaths.PchFilePath, analyzerOptions);
+            var context = new RequestContext(languageKey, rulesConfig, analyzedFilePath, SubProcessFilePaths.PchFilePath, analyzerOptions, CFamilyShared.IsHeaderFileExtension(dbEntry.File));
```

Comment:
it can be in edge conditions(someone compiling their header file) but that definitely not what I meant. It probably should be: `CFamilyShared.IsHeaderFileExtension(analyzedFilePath)`
Very nice catch 👍 

Let me know your preference for the code review. I 'm on another sprint so I might be slower in applying them but I definitely don't mind doing it :D